lib/auth/auth_with_roles.go

```diff
@@ -3968,6 +3991,46 @@ func (a *ServerWithRoles) DeleteDatabaseService(ctx context.Context, name string
 	return a.authServer.DeleteDatabaseService(ctx, name)
 }
 
+// UpsertDiscoveredServer creates or updates a new DiscoveredServer resource.
+func (a *ServerWithRoles) UpsertDiscoveredServer(ctx context.Context, service types.DiscoveredServer) (*types.KeepAlive, error) {
+	if err := a.action(apidefaults.Namespace, types.KindDiscoveredServer, types.VerbCreate, types.VerbUpdate); err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return a.authServer.UpsertDiscoveredServer(ctx, service)
+}
+
+// GetDiscoveredServer gets a DiscoveredServer resource.
+func (a *ServerWithRoles) GetDiscoveredServer(ctx context.Context, instanceID, accountID string) (*types.DiscoveredServerV1, error) {
+	if err := a.action(apidefaults.Namespace, types.KindDiscoveredServer, types.VerbRead); err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return a.authServer.GetDiscoveredServer(ctx, instanceID, accountID)
```

Comment:
Yes, but it's preferable to explicitly invoke the method against the cache for something like this (e.g. `a.authServer.Cache.GetDiscoveredServer(...)`), to better communicate the fact that hitting the cache isn't just preferred, it is required in order to prevent an outage.